include/zephyr/drivers/charger.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2023 Cirrus Logic, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CHARGER_H_
+#define ZEPHYR_INCLUDE_DRIVERS_CHARGER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <zephyr/device.h>
+
+/* Keep these alphabetized wrt property name */
+
+/** Runtime Dynamic Battery Parameters */
+
+/** Indicates if VBUS is present for the charger */
+#define CHARGER_ONLINE	0
+/** Reports whether or not a battery is present */
+#define CHARGER_PRESENT CHARGER_ONLINE + 1
+/** Represents the charging status of the charger */
+#define CHARGER_STATUS	CHARGER_PRESENT + 1
+
+/** Reserved to demark end of common charger properties */
+#define CHARGER_COMMON_COUNT CHARGER_STATUS + 1
+/**
+ * Reserved to demark downstream custom properties - use this value as the actual value may
+ * change over future versions of this API
+ */
+#define CHARGER_CUSTOM_BEGIN CHARGER_COMMON_COUNT + 1
+
+/** Reserved to demark end of valid enum properties */
+#define CHARGER_PROP_MAX UINT16_MAX
+
+enum charger_online {
+	CHARGER_ONLINE_OFFLINE = 0,
+	CHARGER_ONLINE_FIXED,
+	CHARGER_ONLINE_PROGRAMMABLE,
+};
+
+enum charger_status {
+	CHARGER_STATUS_UNKNOWN = 0,
+	CHARGER_STATUS_CHARGING,
+	CHARGER_STATUS_DISCHARGING,
+	CHARGER_STATUS_NOT_CHARGING,
+	CHARGER_STATUS_FULL,
+};
+
+struct charger_get_property {
+	/** Battery charger property to get */
+	uint16_t property_type;
+
+	/** Negative error status set by callee e.g. -ENOTSUP for an unsupported property */
+	int err;
+
+	/** Property field for getting */
+	union {
+		/* Fields have the format: */
+		/* CHARGER_PROPERTY_FIELD */
+		/* type property_field; */
+
+		/* Dynamic Charger Info */
```

Comment:
if necessary to explain details, add this to the struct docs, not inlined.

Reply:
ack